src/main.js

```diff
@@ -325,64 +335,64 @@ function deleteEntity(entities, id) {
   }
 }
 
-function drawNode(node, cx, cy, shape, radius, color) {
+function drawNode(node, cx, cy, shape=defaultShape, radius=defaultRadius, color=defaultColor) {
   let x = parseInt(cx)
   let y = parseInt(cy)
 
   if (shape === "circle") {
-  d3.select(canvas)
-    .append("g")
-    .attr("class", "node")
-    .attr("id", node.id)
-    .attr("transform", "translate("+x+","+y+")")
-    .append(shape)
-    .attr("class", "node-rep")
-    .style("fill", color)
-    .style("z-index", 1)
-    .attr("r", radius)
-    .attr("cx", 0)
-    .attr("cy", 0)
-    .attr("id", node.id)
-    .attr("xmlns", "http://www.w3.org/2000/svg");
+    var nodeG = d3.select(canvas)
+      .append("g")
+      .attr("class", "node")
+      .attr("id", node.id)
+      .attr("transform", "translate("+x+","+y+")");
+    nodeG
+      .append(shape)
+      .attr("class", "node-rep")
+      .style("fill", color)
+      .style("z-index", 1)
+      .attr("r", radius)
+      .attr("cx", 0)
```

Comment:
this function will be depreciated soon since I migrate drawing to a unified redraw based on D3 data model.